src/libostree/ostree-repo-checkout.c

```diff
@@ -912,6 +914,22 @@ checkout_tree_at_recurse (OstreeRepo                        *self,
         return TRUE; /* Note early return */
     }
 
+  if (options->process_whiteouts)
+    {
+      g_autoptr(GVariant) dir_file_contents = g_variant_get_child_value (dirtree, 0);
+      GVariantIter viter;
+      const char *fname;
+      g_autoptr(GVariant) contents_csum_v = NULL;
+      g_variant_iter_init (&viter, dir_file_contents);
+      while (g_variant_iter_loop (&viter, "(&s@ay)", &fname, &contents_csum_v))
+        {
+          is_opaque_whiteout = (g_str_equal (fname, OPAQUE_WHITEOUT_NAME));
+          if (is_opaque_whiteout)
+            break;
```

Comment:
One thing I don't quite understand is why we're looping over all of the children here.  Doesn't there need to be name matching against the target somehow?   Or hmm...is the serialization that there's *exactly one* child with that name?

Reply:
I am looking for a child ".wh..wh..opq" and it needs to have exactly that name.  I didn't find a better way for having this search in O(1).  Is there a better way to look for it?

Reply:
> I am looking for a child ".wh..wh..opq" and it needs to have exactly that name. 

My question is - should that be the *only* child?

Reply:
no, there can be other children and they will appear normally in the directory.  The ".wh..wh..opq" file is only to hide anything that is in the lower layers, but the same layer can add more content